src/main/scala/org/apache/spark/SparkEnv.scala

```diff
@@ -0,0 +1,509 @@
+/*
```

Comment:
As per discussed, let's divide this into two steps: 1. commit Spark files as it is; 2. update our changes. This can reduce a little bit maintenance effort from version to version.

Reply:
Separate PR submitted. Let's merge this one after that one merged.